strings/base_macros.h

```diff
@@ -54,3 +54,27 @@
 struct IUnknown;
 typedef struct _GUID GUID;
 #endif
+
+// std::source_location is a C++20 feature, which is above the C++17 feature floor for cppwinrt.  The source location needs
+// to be the calling code, not cppwinrt itself, so that it is useful to developers building on top of this library.  As a
+// result any public-facing method that can result in an error needs a default-constructed source_location argument.  Because
+// this type does not exist in C++17 we need to use a macro to optionally add parameters and forwarding wherever appropriate.
+#ifdef __cpp_lib_source_location
+#define WINRT_IMPL_SOURCE_LOCATION_ARGS_NO_DEFAULT , std::source_location sourceInformation
+#define WINRT_IMPL_SOURCE_LOCATION_ARGS , std::source_location sourceInformation = std::source_location::current()
+#define WINRT_IMPL_SOURCE_LOCATION_ARGS_SINGLE_PARAM std::source_location sourceInformation = std::source_location::current()
```

Comment:
```suggestion
#define WINRT_IMPL_SOURCE_LOCATION_ARGS , std::source_location const& sourceInformation = std::source_location::current()
#define WINRT_IMPL_SOURCE_LOCATION_ARGS_SINGLE_PARAM std::source_location const& sourceInformation = std::source_location::current()
```
Adding `const&` avoids copying when forwarding. The `source_location` is [rather substantial struct](https://github.com/microsoft/STL/blob/67a96a910fb53aa4bc12801fb9683d30b2647c16/stl/inc/source_location#L53).

Reply:
Good call.  I should have done it that way originally.